auth/auth.go

```diff
@@ -27,20 +27,34 @@ import (
 
 type keyType int
 
+// AuthorizationType designates a type of authorization.
+type AuthorizationType int
+
 type errorResponseHandler func(w http.ResponseWriter, code int, msg string) error
 
 const (
 	// UserIDField is a key
 	UserIDField keyType = iota
 )
 
+const (
+	// AuthorizationTypeNone is the no authorizationType case.
+	AuthorizationTypeNone AuthorizationType = iota
```

Comment:
nit: These are a bit of verbose, perhaps just `None`, `ReadOnly`, `WriteOnly`, and `ReadWrite`?

Reply:
That's fair, in that case perhaps call them `UnknownAuthorization`, `NoAuthorization`, `ReadOnlyAuthorization`, etc.?